packages/account/src/Sections/Security/TwoFactorAuthentication/two-factor-authentication.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { getPropertyValue, WS } from '@deriv/shared';
+import LoadErrorMessage from 'Components/load-error-message';
+import { observer, useStore } from '@deriv/stores';
+import { Loading } from '@deriv/components';
+import TwoFactorEnabled from './two-factor-enabled';
+import TwoFactorDisabled from './two-factor-disabled';
+
+const TwoFactorAuthentication = observer(() => {
+    const { client } = useStore();
+    const { email_address, getTwoFAStatus, has_enabled_two_fa, is_switching } = client;
+
+    const [is_loading, setLoading] = React.useState(true);
```

Comment:
just a thought and you can ignore it:
maybe having so many states we could use useReducer?

Reply:
adding a reducer would be better as we can see the initial values are not being set from props

Even if we need the state to be set from props, we can set the values from an effect